File: first_package/script/minimal_node.py
#! /usr/bin/env python3

import rclpy
from rclpy.node import Node
 
def main(arg=None):
    # 1. Initialize ROS communications for a given context
    rclpy.init(args=arg)
    # 2. Instantiate a Node
    node = Node("minimal_node_py")
    node.get_logger().info(f"Hello from {node.get_name()}")
    # 3. Shutdown a previously initialized context
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: first_package/src/minimal_node.cpp
#include <rclcpp/rclcpp.hpp>


int main(int argc, char* argv[]){
    rclcpp::init(argc, argv);

    rclcpp::Node::SharedPtr node = rclcpp::Node::make_shared("minimal_node_cpp");

    RCLCPP_INFO(node->get_logger(), "Hello from %s", node->get_name());

    rclcpp::shutdown();
}